### ISSUE TITLE
Raspberry Pi4: fatal error: bcm_host.h: No such file or directory

### ISSUE THREAD

**Issue by ZedeN1** [libretro/RetroArch#9727]
## Description

Fatal error when compiling on Raspbian running on Raspberry Pi 4 4Gb.

### Expected behavior

Compiles successfully

### Actual behavior
> CC gfx/drivers/dispmanx_gfx.c
> gfx/drivers/dispmanx_gfx.c:17:10: fatal error: bcm_host.h: No such file or directory
>  #include <bcm_host.h>
>           ^~~~~~~~~~~~
> compilation terminated.
> make: *** [Makefile:213: obj-unix/release/gfx/drivers/dispmanx_gfx.o] Error 1
> make: *** Waiting for unfinished jobs....

./configure output available on [PasteBin](https://pastebin.com/5HA6p8D3)
make -j4 output available on [PasteBin](https://pastebin.com/WtTRkrve)

### Steps to reproduce the bug

1. `git clone https://github.com/libretro/RetroArch.git`
2. `sudo apt-get install build-essential libxkbcommon-dev zlib1g-dev libfreetype6-dev libegl1-mesa-dev libgles2-mesa-dev libgbm-dev libavcodec-dev libsdl2-dev libsdl-image1.2-dev libxml2-dev yasm libavformat-dev libavdevice-dev libswresample-dev libavresample-dev libswscale-dev libv4l-dev libgl*-mesa-dev`
3. ` CFLAGS='-mfpu=neon' ./configure --enable-alsa --enable-udev --enable-floathard --enable-neon --enable-dispmanx --disable-opengl1 --disable-videocore --enable-opengles --enable-opengles3`
4. `make -j4`

### Bisect Results

./configure output available on [PasteBin](https://pastebin.com/5HA6p8D3)
make -j4 output available on [PasteBin](https://pastebin.com/WtTRkrve)

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 666ebb0ae1

### Environment information

- OS: Raspbian on Raspberry Pi 4 4Gb. Linux raspberrypi 4.19.81-v7l+ #1276 SMP Fri Nov 1 14:23:48 GMT 2019 armv7l GNU/Linux]
- Compiler: gcc (Raspbian 8.3.0-6+rpi1) 8.3.0

**Comment by hizzlekizzle** [comment]
Have you tried the solution from this post? https://raspberrypi.stackexchange.com/questions/36121/fatal-error-bcm-host-h-no-such-file-or-directory-compilation-terminated

**Comment by ZedeN1** [comment]
> Have you tried the solution from this post? https://raspberrypi.stackexchange.com/questions/36121/fatal-error-bcm-host-h-no-such-file-or-directory-compilation-terminated

Yes I have, didn't work. running `rpi-update` didn't work either.

I managed to get further in the compile process by adding `-I/opt/vc/include` to CFLAGS but the compile still dies with [error](https://pastebin.com/Bn84Cnah):
```
CC led/drivers/led_rpi.c
LD retroarch
/usr/bin/ld: obj-unix/release/gfx/drivers/dispmanx_gfx.o: undefined reference to symbol 'vc_dispmanx_display_close'
/usr/bin/ld: //opt/vc/lib/libbcm_host.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *** [Makefile:208: retroarch] Error 1
```

**Comment by ZedeN1** [comment]
I was able to compile without dispmanx:

`CFLAGS='-mfpu=neon' ./configure --enable-alsa --enable-udev --enable-floathard --enable-neon --disable-opengl1 --disable-videocore --enable-opengles --enable-opengles3 --disable-dispmanx`

Anyone know why dispmanx is giving me issues?

**Comment by asavah** [comment]
Pi4 does not support or use dispmanx and proprietary (legacy) firmware based GL* drivers (those from /opt/vc).
For pi4 standard opensource mesa drivers should be used over drm/kms, x11 or wayland.
https://github.com/raspberrypi/firmware/issues/1171#issuecomment-507306827

**Comment by zoltanvb** [comment]
Configure scripts should nowadays handle RPi4 without extra options, guide also available: https://docs.libretro.com/guides/rpi/
I believe this one can be closed.

**Comment by LibretroAdmin** [comment]
This should hopefully be resolved by this.
https://github.com/libretro/RetroArch/commit/e859d7cac2cb88e622949ef150777bf38a77de4b
Reopen if there is still an issue
